### PR TITLE
Replace "std::endl" with "\n" (except for error messages)

### PR DESCRIPTION
**Pull request by lucafedeli88**
It is often preferable to use `\n` instead of `std::endl`, since the latter forces the flush of the buffer.
This PR replaces all the occurrences of `std::endl` with `\n` in WarpX.

~**Warning:** while I think that in most cases this is a good idea, I am not sure about error messages. I am inclined to revert to `std::endl` for error messages that are printed just before an abort, to make sure that they are actually printed. What do you think?~

**Note:** Where flushing the buffer seems intentional and useful (e.g., right before an abort or when the message is written to `std::cerr`), I did not replace `std::endl` with `\n`

### CODE REVIEW

**Comment by ax3l** [comment]
> I am inclined to revert to std::endl for error messages [...]

Agreed, please don't generally replace.

**Comment by lucafedeli88** [comment]
> > I am inclined to revert to std::endl for error messages [...]
> 
> Agreed, please don't generally replace.

@ax3l , as agreed I reverted to `std::endl` in cases where flushing the buffer seemed a good idea (essentially for error messages).

File: Source/Diagnostics/BTDiagnostics.cpp [hunk 1]
```diff
@@ -170,18 +170,18 @@ void BTDiagnostics::DerivedInitData ()
...
             warpx.updateStopTime(final_snapshot_fill_time);
             amrex::Print()<<"stop_time insufficient to fill all BTD snapshots. Automatically increased to: "
-                << final_snapshot_fill_time << std::endl;
+                << final_snapshot_fill_time << "\n";
 
         }
         if (warpx.maxStep() == std::numeric_limits<int>::max() && warpx.stopTime() == std::numeric_limits<amrex::Real>::max()) {
             amrex::Print()<<"max_step unspecified and stop time unspecified.  Setting max step to "
-                <<final_snapshot_fill_iteration<< " to fill all BTD snapshots." << std::endl;
+                <<final_snapshot_fill_iteration<< " to fill all BTD snapshots." << "\n";
```
Comment (ax3l) [inline Source/Diagnostics/BTDiagnostics.cpp]:
```suggestion
                <<final_snapshot_fill_iteration<< " to fill all BTD snapshots.\n";
```

File: Source/Diagnostics/FlushFormats/FlushFormatCatalyst.cpp [hunk 2]
```diff
@@ -204,7 +204,7 @@ FlushFormatCatalyst::~FlushFormatCatalyst() {
         amrex::Abort(message);
     } else {
         // Temporary, remove for final
-        std::cout << "Successfully finalized Catalyst" << std::endl;
+        std::cout << "Successfully finalized Catalyst" << "\n";
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatCatalyst.cpp]:
```suggestion
        std::cout << "Successfully finalized Catalyst\n";
```

File: Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp [hunk 3]
```diff
@@ -73,7 +73,7 @@ ChargeOnEB::ChargeOnEB (const std::string& rd_name)
             ofs << "[" << c++ << "]time(s)";
             ofs << m_sep;
             ofs << "[" << c++ << "]Charge (C)";
```
Comment (ax3l) [inline Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp]:
```suggestion
            ofs << "[" << c++ << "]Charge (C)\n";
```

File: Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp [hunk 4]
```diff
@@ -73,7 +73,7 @@ ChargeOnEB::ChargeOnEB (const std::string& rd_name)
...
-            ofs << std::endl;
+            ofs << "\n";
```
Comment (ax3l) [inline Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp]:
```suggestion
```

File: Source/FieldSolver/ImplicitSolvers/SemiImplicitEM.cpp [hunk 5]
```diff
@@ -41,21 +41,21 @@ void SemiImplicitEM::Define ( WarpX*  a_WarpX )
...
     }
     else if (m_nlsolver_type==NonlinearSolverType::Newton) {
-        amrex::Print() << "Nonlinear solver type:      Newton" << std::endl;
+        amrex::Print() << "Nonlinear solver type:      Newton\n";
     }
     m_nlsolver->PrintParams();
-    amrex::Print() << "-----------------------------------------------------------" << std::endl;
-    amrex::Print() << std::endl;
+    amrex::Print() << "-----------------------------------------------------------\n";
+    amrex::Print() << "\n";
```
Comment (ax3l) [inline Source/FieldSolver/ImplicitSolvers/SemiImplicitEM.cpp]:
```suggestion
    amrex::Print() << "-----------------------------------------------------------\n\n";
```

File: Source/FieldSolver/ImplicitSolvers/ThetaImplicitEM.cpp [hunk 6]
```diff
@@ -61,22 +61,22 @@ void ThetaImplicitEM::Define ( WarpX* const  a_WarpX )
...
     }
     else if (m_nlsolver_type==NonlinearSolverType::Newton) {
-        amrex::Print() << "Nonlinear solver type:      Newton" << std::endl;
+        amrex::Print() << "Nonlinear solver type:      Newton\n";
     }
     m_nlsolver->PrintParams();
-    amrex::Print() << "-----------------------------------------------------------" << std::endl;
-    amrex::Print() << std::endl;
+    amrex::Print() << "-----------------------------------------------------------\n";
+    amrex::Print() << "\n";
```
Comment (ax3l) [inline Source/FieldSolver/ImplicitSolvers/ThetaImplicitEM.cpp]:
```suggestion
    amrex::Print() << "-----------------------------------------------------------\n\n";
```

File: Source/Particles/PhysicalParticleContainer.cpp [hunk 7]
```diff
@@ -3259,10 +3259,10 @@ PhysicalParticleContainer::ImplicitPushXP (WarpXParIter& pti,
 #if !defined(AMREX_USE_GPU)
                 std::stringstream convergenceMsg;
                 convergenceMsg << "Picard solver for particle failed to converge after " <<
-                    iter << " iterations. " << std::endl;
+                    iter << " iterations. " << "\n";
```
Comment (ax3l) [inline Source/Particles/PhysicalParticleContainer.cpp]:
```suggestion
                    iter << " iterations.\n";
```

File: Tools/QedTablesUtils/Source/ArgParser/QedTablesArgParser.cpp [hunk 8]
```diff
@@ -85,7 +85,7 @@ ArgParser::ParseArgs (const std::vector<Key>& keys, const int argc, char const*
 void
 ArgParser::PrintHelp (const vector<ArgParser::Key>& cmd_list)
 {
-    cout << "Command line options: " << endl;
+    cout << "Command line options: " << "\n";
```
Comment (ax3l) [inline Tools/QedTablesUtils/Source/ArgParser/QedTablesArgParser.cpp]:
```suggestion
    cout << "Command line options:\n";
```

File: Tools/QedTablesUtils/Source/QedTableGenerator.cpp [hunk 9]
```diff
@@ -49,7 +49,7 @@ void GenerateTableQS (const ParsedArgs& args, const string& outfile_name);
 
 int main (int argc, char** argv)
 {
-    cout << "### QED Table Generator ###" << endl;
+    cout << "### QED Table Generator ###" << "\n";
```
Comment (ax3l) [inline Tools/QedTablesUtils/Source/QedTableGenerator.cpp]:
```suggestion
    cout << "### QED Table Generator ###\n";
```

File: Tools/QedTablesUtils/Source/QedTableReader.cpp [hunk 10]
```diff
@@ -57,7 +57,7 @@ class qs_photon_emission_table_wrapper :
 
 int main (int argc, char** argv)
 {
-    cout << "### QED Table Reader ###" << endl;
+    cout << "### QED Table Reader ###" << "\n";
```
Comment (ax3l) [inline Tools/QedTablesUtils/Source/QedTableReader.cpp]:
```suggestion
    cout << "### QED Table Reader ###\n";
```

**Review by ax3l — COMMENTED** [review]
We can collapse a few extra `<<` between string constants.

**Comment by lucafedeli88** [comment]
> We can collapse a few extra `<<` between string constants.

I agree. I have accepted all your suggestions!

File: Source/Diagnostics/FlushFormats/FlushFormatCatalyst.cpp [hunk 11]
```diff
@@ -204,7 +204,7 @@ FlushFormatCatalyst::~FlushFormatCatalyst() {
         amrex::Abort(message);
     } else {
         // Temporary, remove for final
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatCatalyst.cpp:206]: :D

**Review by ax3l — APPROVED** [review]
Thank you, LGTM :)

**Pull request merged**